File: app/src/test/java/io/oldering/tvfoot/red/DataBindingAdaptersTest.java
package io.oldering.tvfoot.red;

import android.view.View;
import io.oldering.tvfoot.red.app.common.DataBindingAdapters;
import org.junit.Test;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;

public class DataBindingAdaptersTest {
  @Test public void setVisibilityGone() {
    View view = mock(View.class);
    DataBindingAdapters.setVisibility(view, false);
    verify(view).setVisibility(View.GONE);
  }

  @Test public void setVisibilityVisible() {
    View view = mock(View.class);
    DataBindingAdapters.setVisibility(view, true);
    verify(view).setVisibility(View.VISIBLE);
  }
}

File: app/src/test/java/com/benoitquenaudon/tvfoot/red/app/common/DataBindingAdaptersTest.kt
package com.benoitquenaudon.tvfoot.red.app.common

import android.view.View
import android.widget.TextView
import org.junit.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.never
import org.mockito.Mockito.verify

class DataBindingAdaptersTest {
  @Test fun setVisibility_shouldBeGone() {
    val view = mock(View::class.java)
    setVisibility(view, isVisible = false)
    verify(view).visibility = View.GONE
  }

  @Test fun setVisibility_shouldBeVisible() {
    val view = mock(View::class.java)
    setVisibility(view, isVisible = true)
    verify(view).visibility = View.VISIBLE
  }

  @Test fun setDangerText_shouldNotSet() {
    val textView = mock(TextView::class.java)
    setDangerText(textView, textResId = 0)
    verify(textView, never()).setText(0)
  }

  @Test fun setDangerText_shouldSet() {
    val textView = mock(TextView::class.java)
    setDangerText(textView, textResId = 1)
    verify(textView).setText(1)
  }
}
